server/app/src/main/java/com/ams/app/controller/AttendanceController.java

```diff
@@ -0,0 +1,89 @@
+package com.ams.app.controller;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.ams.app.dto.AttendanceReportdto;
+import com.ams.app.dto.Attendrecorddto;
+import com.ams.app.dto.Requestdto;
+
+import com.ams.app.service.AttendrecordService;
+
+@RestController
+@RequestMapping("/attendance")
+public class AttendanceController {
```

Comment:
There are many unnecessary endpoints. Refactor all controller endpoints to follow below pattern
 - GetMapping /index (Which is used to findAll entries with a Specification)
 - GetMapping /{id}     (finds an entry with id)
 - PutMapping /{id}     (takes an object and update it)
 - PostMapping /{id}   (Takes an Object and create a new entry)
 
 
 If something cannot be achieved by these, Then we can create new endpoints

Reply:
Modified all the end points